lib/model/model.go

```diff
@@ -363,8 +363,8 @@ func (m *model) AddFolder(cfg config.FolderConfiguration) {
 	}
 
 	m.fmut.Lock()
+	defer m.fmut.Unlock()
 	m.addFolderLocked(cfg)
-	m.fmut.Unlock()
```

Comment:
Defer prevents inlining, so I am not sure why we thing this is better...

Reply:
I deferred it because of the function call, which incidentally also isn't inlineable: https://github.com/golang/go/wiki/CompilerOptimizations#function-inlining